src/app/blob_pager.rs

```diff
@@ -26,13 +48,42 @@ impl<'repo> BlobPager<'repo> {
             Ok(v) => v,
             Err(e) => panic!("unable to decode utf8 {}", e),
         };
-        let lines = content.lines().map(|line| line.to_string()).collect();
+        let lines: Vec<String> = content.lines().map(|line| line.to_string()).collect();
+        // Load these once at the start of your program
+        let syntax_set = two_face::syntax::extra_newlines();
+        let theme_set = two_face::theme::extra();
+
+        let syntax = match Path::new(&name).extension().and_then(OsStr::to_str) {
+            Some(ext) => syntax_set.find_syntax_by_extension(ext).cloned(),
+            _ => match lines.first() {
+                Some(line) => syntax_set.find_syntax_by_first_line(line).cloned(),
+                _ => None,
+            },
+        };
```

Comment:
While concise, I find this visually hard to parse, or read. Perhaps something like this would be easier to read? What do you think?

```suggestion
        let syntax = {
            let extension = Path::new(&name).extension().and_then(OsStr::to_str);
            if let Some(ext) = extension {
                syntax_set.find_syntax_by_extension(ext).cloned()
            } else if let Some(line) = lines.first() {
                syntax_set.find_syntax_by_first_line(line).cloned()
            } else {
                None
            }
        };
```